pkg/controller/cluster/http_handlers.go

```diff
@@ -116,3 +128,210 @@ func validateBucketName(bucket string) (bool, error) {
 	}
 	return true, nil
 }
+
+// AssumeRoleWithWebIdentityHandler - POST /sts/{tenantNamespace}
+// AssumeRoleWithWebIdentity - implementation of AWS STS API.
+// Authenticates a Kubernetes Service accounts using a JWT Token
+// Evalues a PolicyBinding CRD as Mapping of the Minio Policies that the ServiceAccount can assume on a minio tenant
+// Eg:-
+// $ curl -k -X POST https://operator:9443/sts/{tenantNamespace} -d "Version=2011-06-15&Action=AssumeRoleWithWebIdentity&WebIdentityToken=<jwt>" -H "Content-Type: application/x-www-form-urlencoded"
+func (c *Controller) AssumeRoleWithWebIdentityHandler(w http.ResponseWriter, r *http.Request) {
+	routerVars := mux.Vars(r)
+	tenantNamespace := ""
+	tenantNamespace, err := xhttp.UnescapeQueryPath(routerVars["tenantNamespace"])
+
+	reqInfo := ReqInfo{
+		RequestID:       w.Header().Get(AmzRequestID),
+		RemoteHost:      xhttp.GetSourceIPFromHeaders(r),
+		Host:            r.Host,
+		UserAgent:       r.UserAgent(),
+		API:             webIdentity,
+		TenantNamespace: tenantNamespace,
+	}
+
+	ctx := context.WithValue(r.Context(), contextLogKey, &reqInfo)
+
+	if err != nil {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue, fmt.Errorf("tenant namespace is missing"))
+	}
+
+	// Parse the incoming form data.
+	if err := xhttp.ParseForm(r); err != nil {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue, fmt.Errorf("Error parsing request: %s", err))
+		return
+	}
+
+	if r.Form.Get(stsVersion) != stsAPIVersion {
+		err := fmt.Errorf("invalid STS API version %s, expecting %s", r.Form.Get("Version"), stsAPIVersion)
+		writeSTSErrorResponse(ctx, w, true, ErrSTSMissingParameter, err)
+		return
+	}
+
+	action := r.Form.Get(stsAction)
+	switch action {
+	// For now we only do WebIdentity, leaving it in case we want to implement certificate authentication
+	case webIdentity:
+	default:
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue, fmt.Errorf("unsupported action %s", action))
+		return
+	}
+
+	token := strings.TrimSpace(r.Form.Get(stsWebIdentityToken))
+
+	if token == "" {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSMissingParameter, fmt.Errorf("missing %s", stsWebIdentityToken))
+		return
+	}
+
+	// VALIDATE JWT
+	accessToken := r.Form.Get(stsWebIdentityToken)
+	saAuthResult, err := c.ValidateServiceAccountJWT(&ctx, accessToken)
+	if err != nil {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidIdentityToken, err)
+		return
+	}
+	if !saAuthResult.Status.Authenticated {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSAccessDenied, fmt.Errorf("Access denied: Invalid Token"))
+		return
+	}
+	chunks := strings.Split(strings.Replace(saAuthResult.Status.User.Username, "system:serviceaccount:", "", -1), ":")
+	// saNamespace Service account Namespace
+	saNamespace := chunks[0]
```

Comment:
possible NPE if the split didn't go as planned, should check length is the expected one for `chunks`